agent/internal/binding-accessor/binding-accessor.go

```diff
@@ -115,8 +132,11 @@ func compileField(valueFn valueFunc, buf string) (valueFunc, string, error) {
 		return nil, buf, sqerrors.New("unexpected empty field name")
 	}
 
-	return func(ctx Context) interface{} {
-		return execFieldAccess(valueFn(ctx), field)
+	return func(ctx Context, depth int) interface{} {
+		if depth == 0 {
+			panic(ErrMaxExecutionDepth)
```

Comment:
Should return an error IMHO, especially since you already have an error return type available in the signature.

Reply:
Me being lazy I guess since everything else can panic because of `reflect`. The function literal doesn't have a returned error.

I am reviewing this as I agree it would be better to make it an error since it could definitely happen often.